app/data/tasks/deduplicate_records.py

```diff
@@ -0,0 +1,185 @@
+    else:
+        dscore = ddelta / get_distance_allowance()
+    score = (tscore + dscore) / 2
+    return score
+
+
+def find_duplicates_for_record(uuid):
+    """ Given a record, find all possible duplicates for it
+    """
+    record = Record.objects.get(uuid=uuid)
+    return Record.objects.filter(
+        schema__record_type=record.schema.record_type,
+        geom__dwithin=(record.geom, get_distance_allowance()),
+        occurred_from__range=(
+            record.occurred_from - get_time_allowance(), record.occurred_from + get_time_allowance()
+        )
+    ).exclude(uuid=record.uuid)
+
+
+def get_time_extent():
+    """ Get the time range to dedupe
+    Calculate from X hr before the last dedupe run to the current task run's
+    start time, where X is the heuristic's time allowance
+
+    Returns:
+        dictionary containing the start and end times for the next dedupe task
+        {starttime: timestamp, endtime: timestamp}
+    """
+    # TODO: get config var determining matching time - use value of 1 hr for now
+    time_allowance = get_time_allowance()
+    last_job = None
+    if DedupeJob.objects.all().count() > 0:
+        last_job = DedupeJob.objects.latest()
+    start_time = None
+    if last_job:
+        start_time = last_job.datetime - time_allowance
+    else:
+        start_time = Record.objects.earliest('occurred_from').occurred_from
+    end_time = datetime.datetime.utcnow().replace(tzinfo=timezone('UTC'))
+    return {'start_time': start_time, 'end_time': end_time}
+
+
+def get_dedupe_set(extent):
+    """ Get the set of records to dedupe
+    return a set of uuids to check for duplicates, and the queryset used to
+    generate it
+    """
+    queryset = Record.objects.filter(
+        occurred_from__range=(
+            extent['start_time'], extent['end_time']
+        )
+    ).exclude(uuid__in=get_dedupe_ids())
+    ids = list(queryset.values_list(queryset.model._meta.pk.name, flat=True))
+    return (ids, queryset)
+
+
+def get_dedupe_ids(job=None):
+    """ Get the ids of records to dedupe
+    return a set of uuids to check for duplicates.
+    """
+    queryset = None
+    if job:
+        queryset = RecordDuplicate.objects.filter(job=job)
+    else:
+        queryset = RecordDuplicate.objects.all()
+    ids = list(queryset.values_list('record', flat=True))
+    return ids
+
+
+@shared_task
+def find_duplicate_records():
+    """ Find all duplicate records
+    Duplicate records are classified as two records that occured at the same time
+    and location
+
+    Heuristic:
+    Records that have times within 1 day of each other
+    Records that are located ~100m from each other
+
+    Multiple tasks running:
+    Each job will only consider records added after the last task started.
+    This should allow multiple tasks to run at the same time without identifying
+    duplicates more than once.
+
+    Job failure:
+    On job failure, the task should be retried after deleting the duplicates
```

Comment:
If you need to do cleanup tasks on failure you could use a Celery `errback` to be reasonably sure that the cleanup will occur even if the Celery task fails in a manner that doesn't update the task object status: http://docs.celeryproject.org/en/latest/userguide/calling.html#linking-callbacks-errbacks